charts/helm-dep-installer/templates/_helper.tpl

```diff
@@ -0,0 +1,75 @@
+{{/* vim: set filetype=mustache: */}}
+
+{{/*
+Common labels
+*/}}
+{{- define "generic-dep-installer.labels" -}}
+helm.sh/chart: {{ .Chart.Name }}-{{ .Chart.Version }}
+{{- if .Chart.AppVersion }}
+app.kubernetes.io/version: {{ .Chart.AppVersion | quote }}
+serving.knative.dev/release: {{ .Chart.AppVersion }}
+{{- end }}
+app.kubernetes.io/managed-by: {{ .Release.Service }}
+app.kubernetes.io/part-of: {{ .Release.Name }}
+{{- end }}
+
+{{/*
+Hook annotations
+*/}}
+{{- define "generic-dep-installer.hook-annotations" -}}
+"helm.sh/hook-weight": {{ .Values.hook.weight | default "-5" | quote }}
+"helm.sh/hook": {{ .Values.hook.type | default "pre-install,pre-upgrade" | quote }}
+{{- end }}
+
+{{/*
+Generated names
+*/}}
+{{- define "generic-dep-installer.chart-version" -}}
+{{- $chartVersion := .Values.helmChart.version | replace "." "-" }}
+{{- printf "%s" $chartVersion}}
+{{- end }}
+
+{{- define "generic-dep-installer.resource-prefix" -}}
+{{- $deployedChart := .Values.helmChart.chart -}}
+{{- $chartVersion := .Values.helmChart.version | replace "." "-" }}
+{{- $deployedReleaseName := .Values.helmChart.release -}}
+{{ printf "gdi-%s-%s-%s"  $deployedChart $chartVersion $deployedReleaseName }}
+{{- end }}
+
+{{- define "generic-dep-installer.global-resource-prefix" -}}
+{{- $prefix := include "generic-dep-installer.resource-prefix" . }}
+{{- printf "%s-%s" $prefix .Release.Namespace | trunc 63 | trimSuffix "-" }}
+{{- end }}
+
+{{- define "generic-dep-installer.sa-name" -}}
+{{- printf "%s-sa" (include "generic-dep-installer.resource-prefix" .) | trunc 63 | trimSuffix "-"}}
+{{- end }}
+
+{{- define "generic-dep-installer.job-name" -}}
+{{- printf "%s-job" (include "generic-dep-installer.resource-prefix" .) | trunc 63 | trimSuffix "-"}}
+{{- end }}
+
+{{- define "generic-dep-installer.cm-name" -}}
+{{- printf "%s-cm" (include "generic-dep-installer.resource-prefix" .) | trunc 63 | trimSuffix "-"}}
+{{- end }}
+
+{{- define "generic-dep-installer.job-commands" -}}
+- {{ .Values.helmChart.repository }}
+- {{ .Values.helmChart.chart }}
+- {{ .Values.helmChart.version }}
+- {{ .Values.helmChart.release }}
+- {{ .Values.helmChart.namespace }}
+{{- end }}
+
+{{/*
+Configured hook values
+*/}}
+{{- define "generic-dep-installer.sa-hook-value" -}}
+{{ printf "%s" ((sub .Values.hook.weight 1) | toString) | quote }}
```

Comment:
Just to confirm, don't have to typecast this? if in the parent chart it is configured as a string, would `sub` still work?

Reply:
Just tested with 
```
hook:
  weight: "-5"
```
works as per normal